packages/shopping-cart/src/pages/Cart.jsx

```diff
@@ -0,0 +1,57 @@
+import { useContext } from "react";
+import CartItem from "./CartItem";
+import { arrayOfProducts } from "../products.json";
+import {CartContext} from "../App";
+import { Link } from 'react-router-dom';
+
+function Cart(props) {
+
+  const tableHeader = ["Product", "Name", "Price", "Quantity", "Total", "Remove"]
+  const tableHeaderList = tableHeader.map(header => <th scope="col"> {header} </th>)
+
+  function getProductInfo(item) {
+    return arrayOfProducts.find(product => product.id === item.id)
+  }
+
+  function handleQuantity(quantity, id) {
+    const updatedCart = [...cart];
+    const productIndex = updatedCart.findIndex(item => item.id === id)
+    updatedCart[productIndex].quantity = quantity;
+    updateCart(updatedCart);
+  }
+
+  function handleRemove(id) {
+    const updatedCart = cart.filter(item => item.id !== id)
+    updateCart(updatedCart);
+  }
+
+  const {cart, addToCart, updateCart} = useContext(CartContext);
+
+  
+  return(
+    <div>
+      <div className="table-responsive container table-wrapper">
+        <div className="text-end">
+          <Link to="/checkout" className="btn btn-primary mb-3">Proceed to Checkout <i class="uil uil-angle-right"></i> </Link>
+        </div>
+        <table className="table table-striped table-bordered align-middle">
+          <thead>
+            <tr className="align-middle">
+              {tableHeaderList}
+            </tr>
+          </thead>
+          <tbody>
+            {cart.map(item => {
+              const product = getProductInfo(item)
```

Comment:
can we do this at `componentDidMount` instead of JSX. see component life cycle methods using hooks in case of confusion.

Reply:
not working in useEffect. also I am getting this warning

> Assignments to the 'CartList' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.

Reply:
but doing this works

```suggestion
const CartList = cart.map(item => {
    const product = getProductInfo(item)
    return <CartItem quantity={item.quantity} key={product.id} product={product} handleQuantity= {handleQuantity} handleRemove={handleRemove} />
});
```